pkg/asset/installconfig/clusterid.go

```diff
@@ -74,6 +75,15 @@ func generateInfraID(base string, maxLen int) string {
 	}
 	base = strings.TrimRight(base, "-")
 
+	// Return the InfraID provided by the "INFRA_ID" environment variable if valid
+	if envInfraID, ok := os.LookupEnv("INFRA_ID"); ok && envInfraID != "" {
+		validInfraIDString := fmt.Sprintf("^%s-([b-df-hj-np-tv-z0-9]){%d}$", base, randomLen)
```

Comment:
Maybe move this pattern to a constant and explain what are you looking for

Reply:
I tried to explain more in the comment, moving the pattern to a constant make the code less readable IMO.